src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPGlobalProperties.cs

```diff
@@ -0,0 +1,392 @@
+            string[] v4connections = tcp4FileContents.Split(s_newLineSeparator, StringSplitOptions.RemoveEmptyEntries);
+
+            string tcp6FileContents = File.ReadAllText(NetworkFiles.Tcp6ConnectionsFile);
+            string[] v6connections = tcp6FileContents.Split(s_newLineSeparator, StringSplitOptions.RemoveEmptyEntries);
+
+            TcpConnectionInformation[] connections = new TcpConnectionInformation[v4connections.Length + v6connections.Length - 2]; // First line is header in each file
+            int index = 0;
+
+            // UDP Connections
+            for (int i = 1; i < v4connections.Length; i++) // Skip first line header
+            {
+                string line = v4connections[i];
+                connections[index++] = ParseTcpConnectionInformationFromLine(line);
+            }
+
+            // UDP6 Connections
+            for (int i = 1; i < v6connections.Length; i++) // Skip first line header
+            {
+                string line = v6connections[i];
+                connections[index++] = ParseTcpConnectionInformationFromLine(line);
+            }
+
+            return connections;
+        }
+
+        public override IPEndPoint[] GetActiveTcpListeners()
+        {
+            string tcp4FileContents = File.ReadAllText(NetworkFiles.Tcp4ConnectionsFile);
+            string[] v4connections = tcp4FileContents.Split(s_newLineSeparator, StringSplitOptions.RemoveEmptyEntries);
+
+            string tcp6FileContents = File.ReadAllText(NetworkFiles.Tcp6ConnectionsFile);
+            string[] v6connections = tcp6FileContents.Split(s_newLineSeparator, StringSplitOptions.RemoveEmptyEntries);
+
+            IPEndPoint[] endPoints = new IPEndPoint[v4connections.Length + v6connections.Length - 2]; // First line is header in each file
+            int index = 0;
+
+            // TCP Connections
+            for (int i = 1; i < v4connections.Length; i++) // Skip first line header
+            {
+                string line = v4connections[i];
+                IPAddress remoteIPAddress;
+                int remotePort;
+                ParseRemoteConnectionInformation(line, out remoteIPAddress, out remotePort);
+
+                endPoints[index++] = new IPEndPoint(remoteIPAddress, remotePort);
+            }
+
+            // TCP6 Connections
+            for (int i = 1; i < v6connections.Length; i++) // Skip first line header
+            {
+                string line = v6connections[i];
+                IPAddress remoteIPAddress;
+                int remotePort;
+                ParseRemoteConnectionInformation(line, out remoteIPAddress, out remotePort);
+
+                endPoints[index++] = new IPEndPoint(remoteIPAddress, remotePort);
+            }
+
+            return endPoints;
+        }
+
+        public override IPEndPoint[] GetActiveUdpListeners()
+        {
+            string udp4FileContents = File.ReadAllText(NetworkFiles.Udp4ConnectionsFile);
```

Comment:
I don't remember what we've decided: will this bubble eventual exceptions as System.IO exceptions without converting to NetworkInformationException as we do on Windows?
Unfortunately there aren't sufficient tests to pin behavior where the information isn't accessible (e.g. when the user doesn't have the privilege to access the stats.)